compiler/qsc_frontend/src/compile.rs

```diff
@@ -530,3 +571,20 @@ fn assert_no_errors(sources: &SourceMap, errors: &mut Vec<Error>) {
         panic!("could not compile package");
     }
 }
+
+#[must_use]
+pub fn longest_common_prefix<'a>(strs: &'a [&'a str]) -> &'a str {
```

Comment:
How does this work if it's not taking path separators into account? Won't the common prefix of `/foo/bar` and `/foo/baz` be `/foo/ba`?